src/tests/test_circumcision.py

```diff
@@ -0,0 +1,435 @@
+                    & data[col.CIRCUMCISED]]) == 0
+
+    # no uncircumcised people have circumcision dates
+    assert len(data[~data[col.CIRCUMCISED]
+                    & data[col.CIRCUMCISION_DATE].notnull()]) == 0
+
+    # all circumcised people have circumcision dates
+    assert len(data[data[col.CIRCUMCISED]]) == \
+           len(data[data[col.CIRCUMCISED] & data[col.CIRCUMCISION_DATE].notnull()])
+
+    # check circumcised value is within 3 standard deviations
+    assert mean - 3 * stdev <= no_circumcised <= mean + 3 * stdev
+
+
+def test_vmmc_basic():
+
+    N = 100000
+    start_date = date(2008, 12, 1)
+    stop_date = date(2028, 12, 1)
+    time_step = timedelta(days=90)
+
+    # build artificial population
+    pop = Population(size=N, start_date=start_date)
+    pop.data[col.SEX] = SexType.Male
+    pop.data[col.AGE] = 18
+    pop.data[col.CIRCUMCISED] = False
+    pop.data[col.CIRCUMCISION_DATE] = None
+    pop.data[col.VMMC] = False
+
+    pop.circumcision.circ_inc_rate_year_i = 0
+    pop.circumcision.covid_disrup_affected = 0
+    pop.circumcision.vmmc_disrup_covid = 0
+    pop.circumcision.mc_int = 2008
+    pop.circumcision.year_interv = 2022
+
+    # evolve population
+    pop.data.age += time_step.days / 365
+    pop.circumcision.update_vmmc(pop)
+    # check no VMMC occurs until after mc_int
+    assert len(pop.data[pop.data[col.VMMC]]) == 0
+
+    # change date to appropriate year
+    pop.date += time_step
+    # evolve population
+    pop.data.age += time_step.days / 365
+    pop.circumcision.update_vmmc(pop)
+
+    # get stats
+    no_vmmc = len(pop.data[pop.data[col.VMMC]])
+    male_population = pop.data[(pop.data[col.SEX] == SexType.Male)
+                               & (pop.data[col.AGE] >= 10)
+                               & (pop.data[col.AGE] < 50)]
+    prob_circ = (pop.date.year - pop.circumcision.mc_int) * pop.circumcision.circ_inc_rate
+    mean = len(male_population) * prob_circ
+    stdev = sqrt(mean * (1 - prob_circ))
+    # basic checks
+    general_circumcision_checks(mean, stdev, no_vmmc, pop.data)
+    # nobody over 50 has been circumcised
+    assert len(pop.data[(pop.data[col.AGE] >= 50) & (pop.data[col.VMMC])]) == 0
+
+    # evolve population
+    while pop.date <= stop_date:
+        circ_males = pop.data.index[(pop.data[col.SEX] == SexType.Male) & pop.data[col.CIRCUMCISED]]
+        # advance ages and vmmc
+        pop.data.age += time_step.days / 365
+        pop.circumcision.update_vmmc(pop)
+        pop.date += time_step
+        # check circumcisied people remain circumcised each step
+        new_circ_males = pop.data.index[(pop.data[col.SEX] == SexType.Male)
+                                        & pop.data[col.CIRCUMCISED]]
+        assert circ_males.isin(new_circ_males).all()
+
+    # nobody under 10 has been circumcised
+    assert len(pop.data[(pop.data[col.AGE] < 10) & (pop.data[col.VMMC])]) == 0
+
+
+def test_vmmc_case_i1():
```

Comment:
Would be good if the VMMC tests could be parameterised for different age groups, as we are checking the 15-20 age groups but not 20-30 or 30+